ignite/engine/engine.py

```diff
@@ -791,7 +793,8 @@ def _run_once_on_dataset(self) -> float:
                     # Avoid Events.GET_BATCH_STARTED triggered twice when data iter is restarted
                     if self.last_event_name != Events.DATALOADER_STOP_ITERATION:
                         self._fire_event(Events.GET_BATCH_STARTED)
-                    self.state.batch = next(self._dataloader_iter)
+                    if self._dataloader_iter is not None:
```

Comment:
This condition seems a bit strange too. Let's check and raise an exception before entering while loop.